src/app/shared/users-details/users-details.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { UsersService } from 'src/app/shared/users.service';
+import { ActivatedRoute } from '@angular/router';
+import { User } from '../users.model';
+
+@Component({
+  selector: 'app-users-details',
+  templateUrl: './users-details.component.html',
+  styleUrls: ['./users-details.component.css']
+})
+export class UsersDetailsComponent implements OnInit {
+
+  paramsId: any;
+  user: User;
+
+  constructor(private usersService: UsersService, private route: ActivatedRoute) { }
+
+  ngOnInit(): void {
+    this.paramsId = +this.route.snapshot.params['id'];
+    this.user = this.usersService.find(this.paramsId);
```

Comment:
As you have already subscribed to the route params change below, these two lines can be removed

Reply:
removed these two lines and used the below function

Reply:
ngOnInit(): void {
    this.route.params.subscribe(() => {
      this.user = this.usersService.getUserDetails(+this.route.snapshot.params['id']);
    });
  }